examples/2_intermediate/mta.ipynb

```diff
@@ -244,33 +662,44 @@
     "            self._thread.join()\n",
     "\n",
     "    def _run(self):\n",
-    "        feed = nyct_gtfs.NYCTFeed(\"https://api-endpoint.mta.info/Dataservice/mtagtfsfeeds/nyct%2Fgtfs\", api_key=API_KEY)\n",
+    "        # This is where we will read and process the real-time data feed\n",
+    "        feed = nyct_gtfs.NYCTFeed(\"https://api-endpoint.mta.info/Dataservice/mtagtfsfeeds/nyct%2Fgtfs\", api_key=\"\")\n",
     "\n",
     "        while self._running:\n",
+    "            print(\"----------------------------------------------\")\n",
     "            print(f\"{datetime.utcnow()}: refreshing MTA feed\")\n",
+    "            print(\"----------------------------------------------\")\n",
+    "            print(\"                                     Station     | Line | Direction   | Arrival time\")\n",
     "            feed.refresh()\n",
-    "            # trains will contain all trains underway currently headed to 34St-Penn Station.\n",
-    "            trains = feed.filter_trips(underway=True, headed_for_stop_id=['128', '128N'])\n",
+    "            trains = feed.filter_trips(underway=True, headed_for_stop_id=self.stations)\n",
     "            # tick whenever feed is refreshed\n",
     "            for train in trains:\n",
     "                for update in train.stop_time_updates:\n",
-    "                    if update.stop_id in ['128', '128N']:\n",
-    "                        #self.push_tick(Event(train=(train.route_id, train.headsign_text), update=update.arrival))\n",
+    "                    if update.stop_id in ['128S', '128N']:\n",
```

Comment:
all you need to do is check in self.stations
also rename stations to ._stations, it shouldnt be accessed externally